ivy/src/main/scala/sbt/ivyint/CachedResolutionResolveEngine.scala

```diff
@@ -348,14 +349,59 @@ private[sbt] trait CachedResolutionResolveEngine extends ResolveEngine {
       } map { _.module }
       new ConfigurationReport(rootModuleConf, modules, details, evicted)
     }
+  /**
+   * Returns a touple of (merged org + name combo, newly evicted modules)
+   */
   def mergeOrganizationArtifactReports(rootModuleConf: String, reports0: Vector[OrganizationArtifactReport], os: Vector[IvyOverride], log: Logger): Vector[OrganizationArtifactReport] =
-    (reports0 groupBy { oar => (oar.organization, oar.name) }).toSeq.toVector flatMap {
-      case ((org, name), xs) =>
-        log.debug(s""":::: $rootModuleConf: $org:$name""")
-        if (xs.size < 2) xs
-        else Vector(new OrganizationArtifactReport(org, name, mergeModuleReports(rootModuleConf, xs flatMap { _.modules }, os, log)))
+    {
+      val pairs = (reports0 groupBy { oar => (oar.organization, oar.name) }).toSeq.toVector map {
+        case ((org, name), xs) =>
+          log.debug(s""":::: $rootModuleConf: $org:$name""")
+          if (xs.size < 2) (xs.head, Vector())
+          else
+            mergeModuleReports(rootModuleConf, xs flatMap { _.modules }, os, log) match {
+              case (survivor, newlyEvicted) =>
+                (new OrganizationArtifactReport(org, name, survivor ++ newlyEvicted), newlyEvicted)
+            }
+      }
+      transitivelyEvict(rootModuleConf, pairs map { _._1 }, pairs flatMap { _._2 }, log)
+    }
+  /**
+   * This transitively evicts any non-evicted modules whose only callers are newly evicted.
+   */
+  @tailrec
+  private final def transitivelyEvict(rootModuleConf: String, reports0: Vector[OrganizationArtifactReport],
+    evicted0: Vector[ModuleReport], log: Logger): Vector[OrganizationArtifactReport] =
+    {
+      val em = evicted0 map { _.module }
+      def isTransitivelyEvicted(mr: ModuleReport): Boolean =
+        mr.callers forall { c => em contains { c.caller } }
+      // Ordering of the OrganizationArtifactReport matters
+      val pairs: Vector[(OrganizationArtifactReport, Vector[ModuleReport])] = reports0 map { oar =>
+        val organization = oar.organization
+        val name = oar.name
+        val (affected, unaffected) = oar.modules.toVector partition { mr =>
+          val x = !mr.evicted && mr.problem.isEmpty && isTransitivelyEvicted(mr)
+          if (x) {
+            log.debug(s""":::: transitively evicted $rootModuleConf: $organization:$name""")
+          }
+          x
+        }
+        val newlyEvicted = affected map { _.copy(evicted = true, evictedReason = Some("transitive-evict")) }
```

Comment:
Traditionally Ivy only give you one word response on evicted reason. In this case we know what caused the eviction because there's caller in `ModuleReport`.